presto-plan-checker-router-plugin/src/main/java/com/facebook/presto/router/scheduler/PlanCheckerRouterPluginPrestoClient.java

```diff
@@ -0,0 +1,243 @@
+                queryStatusInfo.getId(),
+                queryStatusInfo.getInfoUri(),
+                queryStatusInfo.getPartialCancelUri(),
+                queryStatusInfo.getNextUri(),
+                queryStatusInfo.getColumns(),
+                null, // if error thrown no data is returned.
+                null,
+                queryStatusInfo.getStats(),
+                newQueryError(queryStatusInfo.getError()),
+                queryStatusInfo.getWarnings(),
+                queryStatusInfo.getUpdateType(),
+                queryStatusInfo.getUpdateCount());
+
+        return QUERY_RESULTS_CODEC.toJson(queryResults);
+    }
+
+    private QueryError newQueryError(QueryError error)
+    {
+        // todo: the getMessage() calls will still return the previous error location
+        ErrorLocation errorLocation = error.getErrorLocation();
+        ErrorLocation newErrorLocation = null;
+        FailureInfo failureInfo = error.getFailureInfo();
+        if (errorLocation != null) {
+            newErrorLocation = new ErrorLocation(
+                    errorLocation.getLineNumber(),
+                    errorLocation.getColumnNumber() - ANALYZE_CALL.length());
+        }
+        return new QueryError(
+                error.getMessage(),
+                error.getSqlState(),
+                error.getErrorCode(),
+                error.getErrorName(),
+                error.getErrorType(),
+                error.isRetriable(),
+                newErrorLocation,
+                new FailureInfo(
+                        failureInfo.getType(),
+                        failureInfo.getMessage(),
+                        failureInfo.getCause(),
+                        failureInfo.getSuppressed(),
+                        failureInfo.getStack(),
+                        newErrorLocation));
+    }
+
+    private ClientSession parseHeadersToClientSession(HttpServletRequest httpServletRequest)
+    {
+        SessionContext sessionContext = new HttpRequestSessionContext(
+                httpServletRequest,
+                new SqlParserOptions());
+
+        return new ClientSession(
+                planCheckerClusterURI,
+                sessionContext.getIdentity().getUser(),
+                sessionContext.getSource(),
+                sessionContext.getTraceToken(),
+                sessionContext.getClientTags(),
+                sessionContext.getClientInfo(),
+                sessionContext.getCatalog(),
+                sessionContext.getSchema(),
+                sessionContext.getTimeZoneId(),
+                Locale.ENGLISH,
+                getResourceEstimates(sessionContext),
+                sessionContext.getSystemProperties(),
+                sessionContext.getPreparedStatements(),
+                sessionContext.getIdentity().getRoles(),
+                sessionContext.getIdentity().getExtraCredentials(),
+                httpServletRequest.getHeader(PRESTO_TRANSACTION_ID),
+                new Duration(2, MINUTES),
```

Comment:
Can we make the the client session timeout configurable ? Users may not want to wait 2 minutes; they may want to fail early